### PR TITLE
Add more doc for optional-test-infra and PR template

### PR DESCRIPTION
**Pull request by PatrickXYS**
The change I made in this PR:

1. Add more documentations for optional-test-infra
2. Create a PR template so that other folks can take advantage of it

/cc @theofpa @Bobgy

### CODE REVIEW

**Comment by PatrickXYS** [comment]
TODO: we might need to add additional Github Workflow to do simple validatation for K8s CRD YAML, in case invalid resource sync with clusters.

**Comment by theofpa** [comment]
The `tekton.dev/v1beta1` doesn't have a `PipelineResource` CR, it's deprecated. We either have to switch back to `tekton.dev/v1alpha1` or remove the `PipelineResources` from the `User` and `GitOps` directories and define them as `workspaces` in the `Tasks`.

**Review by theofpa — APPROVED** [review]

**Comment by PatrickXYS** [comment]
> The tekton.dev/v1beta1 doesn't have a PipelineResource CR, it's deprecated.

Let me check what's the alternative

**Comment by PatrickXYS** [comment]
@theofpa I'll remove Tekton Pipelineresource in a following PR

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+**Which issue is resolved by this Pull Request:**
+Resolves #
+
+**Description of your changes:**
+
+
+**Checklist:**
+<br> If PR related to **Optional-Test-Infra**,
```
Comment (PatrickXYS) [inline .github/pull_request_template.md:8]: @Bobgy  I think it's also useful for Google test-infra related PR

**Comment by PatrickXYS** [comment]
Merge the PR given test passed

**Pull request merged**